components/omega/doc/design/VertCoord.md

```diff
@@ -0,0 +1,292 @@
+                               const Array2DReal &ZMid,
+                               const Array2DReal &LayerThickness,
+                               const Array2DReal &SpecVol,
+                               const Array2DReal &BottomDepth) {
+
+}    
+```
+
+The public `computeGeopotential` will sum together the $z$ height times $g$, the tidal potential, and self attraction and loading:
+```c++
+void VertCoord::computeGeopotential(const Array2DReal &GeopotentialMid,
+                                    const Array2DReal &ZMid,
+                                    const Array2DReal &TidalPotential
+                                    const Array2DReal &SelfAttractionLoading) {
+                                  
+} 
+```
+
+The public `computePStarThickness` will determine the desired mass thickness used for the $p^\star$ vertical coordinate:
+```c++
+void VertCoord::computePStarThickness(const Array2DReal &LayerThicknessPStar,
+                                      const Array2DReal &VertCoordMovementWeights,
+                                      const Array2DReal &RefLayerThickness) {
+                                  
+} 
+```
+
+The private `minMaxLevelEdge` will determine the various vertical loop bounds on edges:
+```c++
+void VertCoord::minMaxLevelEdge(const Array2DI4 &MinLevelEdgeTop,
+                                const Array2DI4 &MaxLevelEdgeTop,
+                                const Array2DI4 &MinLevelEdgeBot,
+                                const Array2DI4 &MaxLevelEdgeBot,
+                                const Array2DI4 &MinLevelCell,
+                                const Array2DI4 &MaxLevelCell,
+                                const Array2DI4 &CellsOnEdge) {
+
+}
+```
+
+The private `minMaxLevelVertex` will determine the various vertical loop bounds on vertices:
+```c++
+void VertCoord::minMaxLevelVertex(const Array2DI4 &MinLevelVertexTop,
+                                  const Array2DI4 &MaxLevelVertexTop,
+                                  const Array2DI4 &MinLevelVertexBot,
+                                  const Array2DI4 &MaxLevelVertexBot,
+                                  const Array2DI4 &MinLevelCell,
+                                  const Array2DI4 &MaxLevelCell,
+                                  const Array2DI4 &CellsOnVertex) {
+                                  
+} 
+```
+
+
+
+#### 4.2.5 Destruction and removal
+
+No operations are needed in the destructor.
+The erase method will remove a named vertical coordinate instance, whereas the clear method will remove all of
+them. 
+Both will call the destructor in the process.
+```c++
+void VertCoord::erase(const std::string &Name);
+void VertCoord::clear();
+```
+
+## Verification and Testing
+
+### Test: 
+Unit tests will be used to test each of the computations (computePressure, computeZHeight, computeGeopotential, computePStarThickness) for a given mass thickness array. Comparison will be made to a ``truth'' vertical mesh.
```

Comment:
Should we also test these routines with a z-level coordinate to make sure bounds and weighting work with spatially variable `minLevelCell` and `maxLevelCell`?